txtorcon/torcontrolprotocol.py

```diff
@@ -356,11 +357,19 @@ def get_info(self, *args):
 
         :return:
             a ``Deferred`` which will callback with a dict containing
-            the keys you asked for. This just inserts ``parse_keywords``
-            in the callback chain; if you want to avoid the parsing
+            the keys you asked for. If you want to avoid the parsing
             into a dict, you can use get_info_raw instead.
         """
-        return self.get_info_raw(*args).addCallback(parse_keywords)
+        lines = yield self.get_info_raw(*args)
+        rtn = {}
+        key = None
+        for line in lines.split('\n'):
+            if line.split('=', 1)[0] in args:
+                key = line.split('=', 1)[0]
+                rtn[key] = line.split('=', 1)[1]
```

Comment:
If we get a response such as,
```
250-a=
foo
one
```
The first ```line.split('=', 1)[1]``` contains a value ```''```. Then the loop takes the next line and adds it to the current value so, the value now becomes, ```'' + '\n' + 'foo'``` to give us ```'\nfoo'```. Hence, the final value we get is, ```'\nfoo\none'```.
Let me know if you think that this works.